builder/build.go

```diff
@@ -63,50 +67,91 @@ func BuildImage(image string, handler string, functionName string, language stri
 			return nil
 		}
 
-		buildOptPackages, err := getBuildOptionPackages(buildOptions, language, langTemplate.BuildOptions)
-		if err != nil {
-			return err
+		if remoteBuilder != "" {
+			tempDir, err := os.MkdirTemp(os.TempDir(), "builder-*")
+			if err != nil {
+				return fmt.Errorf("failed to create temporary directory for %s, error: %w", functionName, err)
+			}
+			defer os.RemoveAll(tempDir)
 
-		}
+			tarPath := path.Join(tempDir, "req.tar")
 
-		dockerBuildVal := dockerBuild{
-			Image:            imageName,
-			NoCache:          nocache,
-			Squash:           squash,
-			HTTPProxy:        os.Getenv("http_proxy"),
-			HTTPSProxy:       os.Getenv("https_proxy"),
-			BuildArgMap:      buildArgMap,
-			BuildOptPackages: buildOptPackages,
-			BuildLabelMap:    buildLabelMap,
-		}
+			if err := makeTar(builderConfig{Image: imageName, BuildArgs: buildArgMap}, path.Join("build", functionName), tarPath); err != nil {
```

Comment:
@welteki 
Now I also add `buildArgMap` in the `builderConfig` for remote-builder, which I was not doing before.
For some reason, `build-arg` flag is not working with me locally, even without `remote-builder`.

Can you try it out now? and inform